File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] ans = new int[2];
        ans[0] = firstsearch(nums,target);
        ans[1] = secondsearch(nums,target);
        return ans;
    }
    private int firstsearch(int []nums,int target){
        int index = -1;
        int start = 0;
        int end = nums.length-1;
        while(start<=end){
            int middle = start+(end-start)/2;
            if(nums[middle]>=target){
                end = middle-1;
            }else{
                start = middle+1;
            }if(nums[middle]==target){
                index = middle;
            }
        }
        return index;
    }
    private int secondsearch(int []nums,int target){
        int index = -1;
        int start = 0;
        int end = nums.length-1;
        while(start<=end){
            int middle = start+(end-start)/2;
            if(nums[middle]<=target){
                start = middle+1;
            }else{
                end = middle-1;
            }if(nums[middle]==target){
                index = middle;
            }
        }
        return index;
    }
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        def helper1(nums,target):
            start = 0
            end = len(nums)-1
            index1 = -1
            while(start<=end):
                mid = (start+end)//2
                if nums[mid] >= target:
                    end = mid-1
                else:
                    start = mid+1
                if nums[mid] == target:
                    index1 = mid
            return index1

        def helper2(nums,target):
            start = 0
            end = len(nums)-1
            index2 = -1
            while(start<=end):
                mid = (start+end)//2
                if nums[mid]<=target:
                    start = mid+1
                else:
                    end = mid-1
                if(nums[mid]==target):
                    index2 = mid
            return index2

        
        ans = [0]*2
        ans[0] = helper1(nums,target)
        ans[1] = helper2(nums,target)
        return ans